client/src/components/User/UserBeaconSettings.vue

```diff
@@ -0,0 +1,273 @@
+            <span class="cursive">___BEACON_PICKUP___</span>. The Beacon database is rebuild every day, this means if
+            you disable the option here, of if we remove the history, or data in the history, the variants will disappear
+            from the Beacon in the next 24h.
+          </p>
+        </div>
+
+
+        <!-- Detailed information about the beacon history -->
+        <div v-if="enabled" class="gray-box">
+          <!-- Case: History does not exist-->
+          <div v-if="!beaconHistory.id" class="flex-row">
+            <div class="no-shrink">
+              No beacon history found
+            </div>
+            <div class="fill"></div>
+            <div class="no-shrink">
+              <b-button @click="createBeaconHistory">Create Beacon History</b-button>
+            </div>
+          </div>
+
+          <!-- Case: History exists -->
+          <div v-if="beaconHistory.id" class="flex-row">
+            <div class="no-shrink">
+              Variant files are searched in your beacon history
+            </div>
+            <div class="fill"></div>
+            <div class="no-shrink">
+              <b-button @click="switchHistory(beaconHistory.id)">Switch to History</b-button>
+            </div>
+          </div>
+        </div>
+
+        <div v-if="enabled">
+          <p>
+            Datasets must fulfill the following conditions in order to be processed
+          </p>
+          <ul>
+            <li>must be VCF or VCF.bzip format</li>
+            <li>must have a human reference assigned to it (e.g. hg19)</li>
+            <li>must define at least one sample</li>
+          </ul>
+        </div>
+      </b-modal>
+    </div>
+  </b-row>
+</template>
+
+<script>
+import store from "store"
+import axios from "axios";
+import Vue from "vue";
+import BootstrapVue from "bootstrap-vue";
+Vue.use(BootstrapVue);
+
+export default {
+  props: {
+    root: {
+      type: String,
+      required: true,
+    },
+    userId: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      enabled: false,
+      beaconHistory: {}
+    };
+  },
+  created() {
+      this.getBeaconHistory();
+  },
+  methods: {
+    switchHistory: async function (historyId) {
+      // history/switch_to_history?hist_id=
+      await store.dispatch("history/setCurrentHistory", historyId);
+    },
+    getBeaconHistory: function () {
+      axios
+          .get(this.root + "api/histories?q=name&qv=___BEACON_PICKUP___")
+          .then((response) => {
+            if (response.data.length > 0) {
+              // always select first match
+              this.beaconHistory = response.data[0];
+
+            }
+          })
+          .catch((error) => {console.log(error.response)});
+    },
+    async createBeaconHistory() {
+      const annotation = "Variant files will be collected from this history if beacon sharing is activated"
+      await axios
```

Comment:
I think you don't need to await or make this method async if you are using the `then`/`catch` callbacks, you either use async/await or the callbacks.
The same for all the other methods below.